src/otaclient_common/retry_task_map.py

```diff
@@ -120,7 +128,10 @@ def _task_done_cb(
 
     def _fut_gen(self, interval: int) -> Generator[Future[Any], Any, None]:
         finished_tasks = 0
-        while True:
+        while finished_tasks == 0 or finished_tasks != self._total_task_num:
+            if self._total_task_num < 0:
+                return
```

Comment:
-1 is set by the tasks dispatcher, when input tasks iterable doesn't yield any task. in this case, we should stop the task execution loop immediately.